zap/src/main/java/org/parosproxy/paros/core/scanner/AppParameter.java

```diff
@@ -0,0 +1,41 @@
+package org.parosproxy.paros.core.scanner;
+
+/**
+ * {@code AppParameter} class wraps the parameters which are used to modify the {@code HttpMessage}.
+ * It is specifically used for updating multiple parameters of {@code HttpMessage}
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class AppParameter {
+    private NameValuePair nameValuePair;
+    private String param;
+    private String value;
+    private AppParamValueType appParamValueType;
+
+    AppParameter(
+            NameValuePair nameValuePair,
```

Comment:
I'm not sure this needs to be passed around, the variants will have them already (this just needs to provide the position).

Reply:
shouldn't we align with the existing APIs because for a single call we are passing NameValuePair? Also, I think there is no need for the caller to know these internal things?

Reply:
The only user of the constructor is the builder and the consumers are the variants, for the builder it doesn't matter if it passes the position or the whole `NameValuePair`, the variants just need to know about the position.

Reply:
hmm right, i think same reason like getName api.